mtags/src/main/scala/scala/meta/internal/mtags/JavaToplevelMtags.scala

```diff
@@ -170,6 +170,8 @@ class JavaToplevelMtags(
       reader.nextChar()
       reader.ch match {
         case `quote` => Token.Literal
+        case _ if reader.endCharOffset >= reader.buf.length =>
```

Comment:
Is the order of this case in the pattern match important? I wonder if it makes sense to separate a happy path from the unhappy one by e.g. extracting this check earlier, before the pattern match : 🤔

Reply:
Moved it earlier, wasn't neccessary in the pattern match